common/remote/sockfile.cpp

```diff
@@ -3430,6 +3414,34 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
 #endif
     }
 
+    bool takeThrottleSem()
+    {
+        bool got = false;
+        CCycleTimer timer;
+        loop {
+            if (throttlesem.wait(throttleDelayMs)) {
+                got = true;
+                break;
+            }
+            unsigned cpu = getLatestCPUUsage();
+            PROGLOG("Throttler: transaction delayed (%d%% cpu)", cpu);
+
+            // NB: getLatestCPUUsage() is based on interval monitoring, typically 60 secs
+            if (getLatestCPUUsage()<throttleCPULimit)
```

Comment:
yes probably.. although it's cached in the call so makes no practical difference - apart from being non-obvious why coded like this.
Will change.